lib/intercom/request.rb

```diff
@@ -58,11 +62,12 @@ def client(uri, read_timeout:, open_timeout:)
       net
     end
 
-    def execute(target_base_url=nil, username:, secret: nil, read_timeout: 90, open_timeout: 30)
+    def execute(target_base_url=nil, username:, secret: nil, read_timeout: 90, open_timeout: 30, api_version: nil)
       retries = 3
       base_uri = URI.parse(target_base_url)
       set_common_headers(net_http_method, base_uri)
       set_basic_auth(net_http_method, username, secret)
+      set_api_version(net_http_method, api_version) if api_version
```

Comment:
I would rather check if api version is present here: for example an empty string (`api_version = ""`) evaluates to true

```suggestion
      set_api_version(net_http_method, api_version) if api_version.present?
```

Reply:
@GabrielAnca the reason I didn't check for an empty string is that we actually catch that as part of the initial API version validation in the client itself.

I added a couple more tests to make this more clear (see faa92b4).

That said, I can easily add a check here as well if you prefer (although I think we'll want to use `unless api_version&.empty?` instead of relying on `present?` being available)

Reply:
@GabrielAnca just checking in on this, would love to get this PR in so I can switch back to the official gem (and off of my fork).